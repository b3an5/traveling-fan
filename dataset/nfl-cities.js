const nflTeams = [
  {
    "name": "Arizona Cardinals",
    "state": "Arizona",
    "city": "Glendale",
    "division": "NFC West"
  },
  {
    "name": "Atlanta Falcons",
    "state": "Georgia",
    "city": "Atlanta",
    "division": "NFC South"
  },
  {
    "name": "Baltimore Ravens",
    "state": "Maryland",
    "city": "Baltimore",
    "division": "AFC North"
  },
  {
    "name": "Buffalo Bills",
    "state": "New York",
    "city": "Orchard Park",
    "division": "AFC East"
  },
  {
    "name": "Carolina Panthers",
    "state": "North Carolina",
    "city": "Charlotte",
    "division": "NFC South"
  },
  {
    "name": "Chicago Bears",
    "state": "Illinois",
    "city": "Chicago",
    "division": "NFC North"
  },
  {
    "name": "Cincinnati Bengals",
    "state": "Ohio",
    "city": "Cincinnati",
    "division": "AFC North"
  },
  {
    "name": "Cleveland Browns",
    "state": "Ohio",
    "city": "Cleveland",
    "division": "AFC North"
  },
  {
    "name": "Dallas Cowboys",
    "state": "Texas",
    "city": "Frisco",
    "division": "NFC East"
  },
  {
    "name": "Denver Broncos",
    "state": "Colorado",
    "city": "Denver",
    "division": "AFC West"
  },
  {
    "name": "Detroit Lions",
    "state": "Michigan",
    "city": "Detroit",
    "division": "NFC North"
  },
  {
    "name": "Green Bay Packers",
    "state": "Wisconsin",
    "city": "Green Bay",
    "division": "NFC North"
  },
  {
    "name": "Houston Texans",
    "state": "Texas",
    "city": "Houston",
    "division": "AFC South"
  },
  {
    "name": "Indianapolis Colts",
    "state": "Indiana",
    "city": "Indianapolis",
    "division": "AFC South"
  },
  {
    "name": "Jacksonville Jaguars",
    "state": "Florida",
    "city": "Jacksonville",
    "division": "AFC South"
  },
  {
    "name": "Kansas City Chiefs",
    "state": "Kansas",
    "city": "Kansas City",
    "division": "AFC West"
  },
  {
    "name": "Miami Dolphins",
    "state": "Florida",
    "city": "Miami",
    "division": "AFC East"
  },
  {
    "name": "Minnesota Vikings",
    "state": "Minnesota",
    "city": "Minneapolis",
    "division": "NFC North"
  },
  {
    "name": "New England Patriots",
    "state": "Massachusetts",
    "city": "Foxborough",
    "division": "AFC East"
  },
  {
    "name": "New Orleans Saints",
    "state": "Louisiana",
    "city": "New Orleans",
    "division": "NFC South"
  },
  {
    "name": "New York Giants",
    "state": "New York",
    "city": "East Rutherford",
    "division": "NFC East"
  },
  {
    "name": "New York Jets",
    "state": "New York",
    "city": "East Rutherford",
    "division": "AFC East"
  },
  {
    "name": "Oakland Raiders",
    "state": "California",
    "city": "Oakland",
    "division": "AFC West"
  },
  {
    "name": "Philadelphia Eagles",
    "state": "Pennsylvania",
    "city": "Philadelphia",
    "division": "NFC East"
  },
  {
    "name": "Pittsburgh Steelers",
    "state": "Pennsylvania",
    "city": "Pittsburgh",
    "division": "AFC North"
  },
  {
    "name": "St. Louis Rams",
    "state": "Missouri",
    "city": "St. Louis",
    "division": "NFC West"
  },
  {
    "name": "San Diego Chargers",
    "state": "California",
    "city": "San Diego",
    "division": "AFC West"
  },
  {
    "name": "San Francisco 49ers",
    "state": "California",
    "city": "San Francisco",
    "division": "NFC West"
  },
  {
    "name": "Seattle Seahawks",
    "state": "Washington",
    "city": "Seattle",
    "division": "NFC West"
  },
  {
    "name": "Tampa Bay Buccaneers",
    "state": "Florida",
    "city": "Tampa",
    "division": "NFC South"
  },
  {
    "name": "Tennessee Titans",
    "state": "Tennessee",
    "city": "Nashville",
    "division": "AFC South"
  },
  {
    "name": "Washington Redskins",
    "state": "Maryland",
    "city": "Ashburn",
    "division": "NFC East"
  }
]

const cities = [
  {
    "name": "Glendale",
    "state": "Arizona",
    "population": 226721,
    "attractions": ["Wildlife World Zoo", "White Tank Mountain Regional Park", "Westgate Arena", "Thunderbird Conservation Park"]
  },
  {
    "name": "Atlanta",
    "state": "Georgia",
    "population": 486290,
    "attractions": ["Georgia Aquarium", "World of Coca-Cola", "Centennial Olympic Park", "Atlanta Botanic Gardens"]
  },
  {
    "name": "Baltimore",
    "state": "Maryland",
    "population": 611648,
    "attractions": ["Inner Harbor", "National Aquarium", "Fort McHenry", "B&O Railroad Museum"]
  },
  {
    "name": "Orchard Park",
    "state": "New York",
    "population": 29054,
    "attractions": ["Eternal Flame Falls", "Chestnut Ridge Park", "Bob-O-Link Golf Club", "Spa at Falling Waters"]
  },
  {
    "name": "Charlotte",
    "state": "North Carolina",
    "population": 2474314,
    "attractions": ["Carowinds", "NASCAR Hall of Fame", "Discovery Place Science", "U.S. National Whitewater Center"]
  },
  {
    "name": "Chicago",
    "state": "Illinois",
    "population": 2716450,
    "attractions": ["Willis Tower", "Millennium Park", "Cloud Gate", "Navy Pier"]
  },
  {
    "name": "Cincinnati",
    "state": "Ohio",
    "population": 2172191,
    "attractions": ["Cincinnati Zoo", "Ohio River", "Newport Aquarium", "Cincinnati Museum Center"]
  },
  {
    "name": "Cleveland",
    "state": "Ohio",
    "population": 388072,
    "attractions": ["Rock and Roll Hall of Fame", "Cleveland Museum of Art", "Cleveland Metroparks Zoo"]
  },
  {
    "name": "Frisco",
    "state": "Texas",
    "population": 116989,
    "attractions": ["Arbor Hills Nature Preserve", "Frisco Commons Park", "Hope Park", "Karya Siddhi Hanuman Temple"]
  },
  {
    "name": "Denver",
    "state": "Colorado",
    "population": 704621,
    "attractions": ["LoDo", "Denver Botanic Gardens", "Red Rocks Amphitheatre", "Denver Zoo"]
  },
  {
    "name": "Detroit",
    "state": "Michigan",
    "population": 673104,
    "attractions": ["The Henry Ford Museum", "Belle Isle Park", "Renaissance Center", "Detroit Institute of Art"]
  },
  {
    "name": "Green Bay",
    "state": "Wisconsin",
    "population": 104057,
    "attractions": ["Bay Beach Amusement Park", "Green Bay Botanical Garden", "National Railroad Museum", "NEW Zoo and Adventure Park"]
  },
  {
    "name": "Houston",
    "state": "Texas",
    "population": 2312717,
    "attractions": ["Space Center Houston", "Houston Zoo", "Johnson Space Center", "Houston Museum District"]
  },
  {
    "name": "Indianapolis",
    "state": "Indiana",
    "population": 2411086,
    "attractions": ["The Children's Museum of Indianapolis", "White River State Park", "Indianapolis Motor Speedway", "Indianapolis Zoo"]
  },
  {
    "name": "Jacksonville",
    "state": "Florida",
    "population": 892062,
    "attractions": ["Amelia Island", "Jacksonville Zoo and Gardens", "Kathryn Abbey Hanna Park", "Kingsley Plantation"]
  },
  {
    "name": "Kansas City",
    "state": "Kansas",
    "population": 145786,
    "attractions": ["Schlitterbahn Waterpark", "Shawnee Mission Park", "Harrah's North Kansas City", "Loose Park Rose Garden"]
  },
  {
    "name": "Miami",
    "state": "Florida",
    "population": 463347,
    "attractions": ["South Beach", "Ocean Drive", "Bayside Marketplace", "Miami Seaquarium"]
  },
  {
    "name": "Minneapolis",
    "state": "Minnesota",
    "population": 422331,
    "attractions": ["Minnehaha Park", "Walker Art Center", "Minneapolis Sculpture Garden", "Mill City Museum"]
  },
  {
    "name": "Foxborough",
    "state": "Massachusetts",
    "population": 16693,
    "attractions": ["Patriots Place", "F. Gilbert Hills State Forest", "Borderland State Park", "The Nature Trail and Cranberry Bog"]
  },
  {
    "name": "New Orleans",
    "state": "Louisiana",
    "population": 393292,
    "attractions": ["French Quarter", "Bourbon Street", "Garden District", "Jackson Square"]
  },
  {
    "name": "East Rutherford",
    "state": "New Jersey",
    "population": 9856,
    "attractions": ["MetLife Stadium", "Meadowlands Sports Complex", "Meadowlands Museum", "Carlstadt Historical Society Museum"]
  },
  {
    "name": "Oakland",
    "state": "California",
    "population": 425195,
    "attractions": ["Angel Island", "Tilden Regional Park", "Lake Merritt", "Redwood Regional Park"]
  },
  {
    "name": "Philadelphia",
    "state": "Pennsylvania",
    "population": 1580863,
    "attractions": ["Independence Hall", "Liberty Bell", "Philadelphia Museum of Art", "Independence National Historical Park"]
  },
  {
    "name": "Pittsburgh",
    "state": "Pennsylvania",
    "population": 2353045,
    "attractions": ["Downtown Pittsburgh", "Phipps Conservatory", "Pittsburgh Zoo", "The Andy Warhol Museum"]
  },
  {
    "name": "St. Louis",
    "state": "Missouri",
    "population": 308626,
    "attractions": ["Gateway Arch", "Gateway Arch Museum", "Missouri Botannical Gardens", "City Museum"]
  },
  {
    "name": "San Diego",
    "state": "California",
    "population": 1419516,
    "attractions": ["Balboa Park", "SeaWorld", "San Diego Zoo", "USS Midway Museum"]
  },
  {
    "name": "San Francisco",
    "state": "California",
    "population": 884363,
    "attractions": ["Golden Gate Bridge", "Fisherman's Wharf", "Alcatraz Island", "Golden Gate Park"]
  },
  {
    "name": "Seattle",
    "state": "Washington",
    "population": 730410,
    "attractions": ["Space Needle", "Pike Place Market", "Chihuly Garden and Glass", "Museum of Pop Culture"]
  },
  {
    "name": "Tampa",
    "state": "Florida",
    "population": 385430,
    "attractions": ["Busch Gardens", "Ybor City", "The Florida Museum", "ZooTampa at Lowry Park"]
  },
  {
    "name": "Nashville",
    "state": "Tennessee",
    "population": 691243,
    "attractions": ["Grand Ole Opry", "The Parthenon", "Ryman Auditorium", "Country Music Hall of Fame"]
  },
  {
    "name": "Landover",
    "state": "Maryland",
    "population": 23078,
    "attractions": ["United States National Arboretum", "Kenilworth Park and Aquatic Gardens", "United States Capitol", "Greenbelt Park"]
  }
]


module.exports = {
  nflTeams,
  cities
}