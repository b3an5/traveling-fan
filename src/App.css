
.App {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
}

.app-title-area {
  height: 250px;
  opacity: 0.7;
  background: url('https://cdn-s3.si.com/styles/marquee_large_2x/s3/images/philly-eagles-broncos-fans.jpg?itok=1pEiQnWa');
}

.app-title {
  font-size: 3rem;
  font-weight: 800;
  text-align: center;
  padding-top: 30px;
  color: black;
  opacity: 1;
}


@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
