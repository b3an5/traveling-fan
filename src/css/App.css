.App {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100vh;
}

.app-img {
  height: 6rem;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
}

.app-title-area {
  height: 175px;
  /* opacity: 0.7; */
  background: linear-gradient( rgba(255, 255, 255, 0.3), rgba(255, 255, 255, 0.3)), url('../images/philly-eagles-broncos-fans.jpg');
}

.app-title {
  font-size: 6rem;
  font-weight: 800;
  text-align: center;
  padding-top: 30px;
  color: black;
  opacity: 1;
}

.loading-screen {
  display: flex;
  height: 100vh;
  width: 100vw;
}

.loading-image {
  width: 100%;
  justify-content: center;
  align-items: center
}

.icon-img {
  max-height: 100%;
  max-width: 100%;
}

.nav-area {
  background: url('../images/nav5.jpg');
  height: 120px;
  /* opacity: .2; */
}


@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
