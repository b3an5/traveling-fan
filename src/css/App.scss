.App {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100vh;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
}

.app-img {
  height: 6rem;
}

.app-title {
  font-size: 6rem;
  font-weight: 800;
  text-align: center;
  padding-top: 30px;
  color: black;
  opacity: 1;
}

.app-title-area {
  height: 175px;
  opacity: 0.7;
  background: linear-gradient( rgba(255, 255, 255, 0.3), rgba(255, 255, 255, 0.3)), url('../images/philly-eagles-broncos-fans.jpg');
}

.icon-img {
  max-height: 100%;
  max-width: 100%;
}

.loading-image {
  width: 100%;
  justify-content: center;
  align-items: center
}

.loading-screen {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-image: url("https://slack-imgs.com/?c=1&url=http%3A%2F%2Fi.imgur.com%2F4Unb7.jpg");
  background-size: 100%;
  height: 100vh;
  width: 100vw;
}

.nav-area {
  background: url('../images/nav.jpg');
  height: 120px;
  opacity: .8;
}

.welcome-button {
  width: 30%;
  height: 10%;
  border: 6px solid white;
  border-radius: 56px;
  font-size: 3em;
  color: white;
  background-color: black;
  &:hover {
    border: 6px solid black;
    color: black;
    background-color: white;
    cursor: pointer;
    transition: .2s
  }
}

.welcome-text {
  background: linear-gradient(#0e69ae, #fff);
  border: 2px solid black;
  border-radius: 30px;
  text-align: center;
  width: 30%;
  margin-bottom: 20%;
  margin-top: 13px;
}

.welcome-title {
  font-size: 3em;
  padding-bottom: 5px;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}