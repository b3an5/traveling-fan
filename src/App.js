import React, { Component } from 'react';
import { nflTeams, cities } from './dataset/nfl-cities'
import Container from './Container.js'
// import './reset.css';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      nflTeams: null,
      isLoading: true
    }
  }

  componentDidMount() {
    fetch('https://whateverly-datasets.herokuapp.com/api/v1/nflTeams')
      .then(nflTeams => nflTeams.json())
      .then(data => {
        setTimeout(() => {
          this.setState({
            nflTeams: data.nflTeams,
            isLoading: false
          })
        }, 1000);
      }
      )
      .catch(error => console.log(error))
  }

  render() {
    if(this.state.isLoading) {
      return(
        <div className="loading-screen"><img className="loading-image" src='https://media.giphy.com/media/lXiRCXQ2OWXO3xPr2/giphy.gif' /></div>
      )
    } else {
      return (
        <div className="app">
          <div className='app-title-area'>
            <h1 className='app-title'>The Traveling Fan</h1>
          </div>
          {/* <Nav /> */}
          <div className="flex">
          {console.log(this.state.nflTeams)}
            <Container nflTeams={this.state.nflTeams} cities={cities} />
          </div>
        </div>
      );
    }
  }
}

export default App;
