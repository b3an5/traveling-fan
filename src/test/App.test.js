import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import { shallow } from 'enzyme';
import { nflTeams, cities } from '../dataset/nfl-cities.js'

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <App  />
    )
  })

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('should match the snapshot with all data passed correctly to the Popup', () => {
      expect(wrapper).toMatchSnapshot();
  })

  it('should have a default state', () => {
    expect(wrapper.state()).toEqual({
      nflTeams: null,
      cities: null,
      isLoading: true,
      allTeams: null
    })
  })

  it.skip('should insert missing cities into Cities array to match length of NFLTeams array', () => {
    const length = cities.length
    wrapper.instance().matchCitiesToTeams(cities)
    expect(length).toEqual(length + 2)
   /* 
         This feels like something that we should test.
          We are modifying our data in our application
          Should we test against the fetch call
          or test against our data?  ( which is NOW UPDATED WITH 
          ALL cities because we needed to write a test for carousel)
    */
  })

  it('should be able to find a team that has been searched for', () => {
    expect(wrapper.state()).toEqual({
      nflTeams: null,
      cities: null,
      isLoading: true,
      allTeams: null
    })
    wrapper.instance().findSearchedTeam('denver')
    expect(wrapper.state()).toEqual({nflTeams: nflTeams[9]})
  })

  it.skip('should be able to find a selected division/conference', () => {
    // don't forget to write this test, yo!
  })
  
});
