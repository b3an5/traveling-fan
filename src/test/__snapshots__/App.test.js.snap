// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed correctly to the Container 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="welcome-text"
      >
        <h2
          className="welcome-title"
        >
          The Traveling Fan
        </h2>
      </div>,
      "className": "loading-screen",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h2
          className="welcome-title"
        >
          The Traveling Fan
        </h2>,
        "className": "welcome-text",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "The Traveling Fan",
          "className": "welcome-title",
        },
        "ref": null,
        "rendered": "The Traveling Fan",
        "type": "h2",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="welcome-text"
        >
          <h2
            className="welcome-title"
          >
            The Traveling Fan
          </h2>
        </div>,
        "className": "loading-screen",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2
            className="welcome-title"
          >
            The Traveling Fan
          </h2>,
          "className": "welcome-text",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The Traveling Fan",
            "className": "welcome-title",
          },
          "ref": null,
          "rendered": "The Traveling Fan",
          "type": "h2",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
